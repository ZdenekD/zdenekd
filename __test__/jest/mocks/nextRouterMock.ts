const nextRouterMock = jest.mock('next/router', () => ({
    useRouter: jest.fn().mockImplementation(() => ({
        asPath: '/page/subpage/something-else',
        pathname: '/',
        route: '/',
        locale: 'cs',
        locales: ['cs', 'en'],
        query: {param: 'param'},
    })),
}));

export default nextRouterMock;
