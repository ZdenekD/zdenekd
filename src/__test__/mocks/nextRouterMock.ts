const nextRouterMock = jest.mock('next/router', () => ({
    useRouter: jest.fn().mockImplementation(() => ({
        pathname: '/',
        route: '/',
        locale: 'en',
        locales: ['cs', 'en'],
        query: {param: 'param'},
    })),
}));

export default nextRouterMock;
