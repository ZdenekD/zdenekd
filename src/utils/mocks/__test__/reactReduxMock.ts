const reactReduxMock = jest.mock('react-redux', () => ({
    useSelector: () => ({}),
    useDispatch: jest.fn(),
}));

export default reactReduxMock;
