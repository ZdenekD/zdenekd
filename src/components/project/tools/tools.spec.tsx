import {shallow} from 'enzyme';
import findComponent from '@/__test__/utils/helpers';
import '@/mocks/__test__/nextRouterMock';
import Tools from '.';

const items = ['string1', 'string2'];

describe('Project/Tools', () => {
    it('renders without error', () => {
        const wrapper = shallow(<Tools items={items} />);
        const component = findComponent(wrapper, 'component-tools');

        expect(component.exists()).toBe(true);
    });
});
