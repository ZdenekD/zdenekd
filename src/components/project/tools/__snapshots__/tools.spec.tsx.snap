// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tools match snapshot 1`] = `
<ul class="tools"
    data-test="component-tools"
>
  <li class="tool">
    <figure class="figure">
      <img src="/logo_string1.svg"
           width="32"
           height="32"
           alt="Technology used in the project: string1"
           class="image"
      >
      <figcaption class="caption">
        string1
      </figcaption>
    </figure>
  </li>
  <li class="tool">
    <figure class="figure">
      <img src="/logo_string2.svg"
           width="32"
           height="32"
           alt="Technology used in the project: string2"
           class="image"
      >
      <figcaption class="caption">
        string2
      </figcaption>
    </figure>
  </li>
</ul>
`;
